src/client.rs

```diff
@@ -909,7 +909,7 @@ where
                             current_depth + 1,
                             target,
                             snapshot,
-                            Some(meta.as_ref()),
+                            Some((meta.as_ref(), delegation.role().clone())),
```

Comment:
Am I using the right role here?  I think so, but the integration test also wouldn't catch a mistake here.

Reply:
Yeah I think this is the right role.